### PR TITLE
[BugFixed] fix wrong trunc_normal_init use

### PR DESCRIPTION
**Pull request by vealocia**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Implementation of SwinTransformer and PVT use trunc_normal_init incorrectly which may decrease the performance when training from scratch.

## Modification

The use of trunc_normal_init of SwinTransformer and PVT

## Checklist

- [x] Pre-commit or other linting tools are used to fix the potential lint issues.
- [x] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
- [x] The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by AronLin** [comment]
Thanks for your contribution. We have assigned the reviewer and we will give you some comments ASAP.

File: mmdet/models/backbones/pvt.py [hunk 1]
```diff
@@ -6,7 +6,8 @@
 import torch.nn as nn
 import torch.nn.functional as F
 from mmcv.cnn import (Conv2d, build_activation_layer, build_norm_layer,
-                      constant_init, normal_init, trunc_normal_init)
+                      constant_init, normal_init, trunc_normal_,
```
Comment (RangiLyu) [inline mmdet/models/backbones/pvt.py:9]: `trunc_normal_` cannot be imported from `mmcv.cnn`, please import it from `mmcv.cnn.utils.weight_init ` or use `nn.init.trunc_normal_ ` instead.

Comment (vealocia) [reply mmdet/models/backbones/pvt.py]: resolved.

**Review by RangiLyu — APPROVED** [review]
LGTM

File: mmdet/models/backbones/pvt.py [hunk 2]
```diff
@@ -275,7 +276,7 @@ def __init__(self, pos_shape, pos_dim, drop_rate=0., init_cfg=None):
         self.drop = nn.Dropout(p=drop_rate)
 
     def init_weights(self):
-        trunc_normal_init(self.pos_embed, std=0.02)
+        trunc_normal_(self.pos_embed, std=0.02)
```
Comment (hhaAndroid) [inline mmdet/models/backbones/pvt.py:279]: What is the difference between the two?

Comment (vealocia) [reply mmdet/models/backbones/pvt.py:279]: ```trunc_normal_init``` requires input module must be ```nn.Module```, neither ```torch.Tensor``` nor ```nn.Parameter```. If the input is ```torch.Tensor```, the tensor will not be init as it should be.

Comment (vealocia) [reply mmdet/models/backbones/pvt.py:279]: for more details, please refer to [```trunc_normal_init```](https://github.com/open-mmlab/mmcv/blob/c9d464c36094f062add8183d5e4bcef570f83cbc/mmcv/cnn/utils/weight_init.py#L73)

Comment (RangiLyu) [reply mmdet/models/backbones/pvt.py:279]: > What is the difference between the two?

 `trunc_normal_init` can only init module with weight and bias, not nn.Parameter. The position embedding is not initialized correctly now.

Comment (hhaAndroid) [reply mmdet/models/backbones/pvt.py:279]: O, I get it. It is best to add a note to explain the precautions, because it is very easy to confuse.

**Review by hhaAndroid — APPROVED** [review]

File: mmdet/models/backbones/pvt.py (hunk 2 above)
Comment (vealocia) [reply mmdet/models/backbones/pvt.py:279]: Maybe it's better to add some warning in ```MMCV```, two implementations of ```trunc init``` are really really confused.

Comment (RangiLyu) [reply mmdet/models/backbones/pvt.py:279]: > Maybe it's better to add some warning in `MMCV`, two implementations of `trunc init` are really really confused.

Agree. Welcome to create a PR to MMCV.

File: mmdet/models/backbones/swin.py [hunk 3]
```diff
@@ -672,12 +673,10 @@ def init_weights(self):
...
+                trunc_normal_(self.absolute_pos_embed, std=0.02)
             for m in self.modules():
                 if isinstance(m, nn.Linear):
-                    trunc_normal_init(m.weight, std=.02)
-                    if m.bias is not None:
-                        constant_init(m.bias, 0)
+                    trunc_normal_init(m, std=.02, bias=0.)
                 elif isinstance(m, nn.LayerNorm):
                     constant_init(m.bias, 0)
                     constant_init(m.weight, 1.0)
```
Comment (Junjun2016) [inline mmdet/models/backbones/swin.py:682]: constant_init is also for modules

File: mmdet/models/backbones/pvt.py [hunk 4]
```diff
@@ -486,9 +487,7 @@ def init_weights(self):
                         f'training start from scratch')
             for m in self.modules():
                 if isinstance(m, nn.Linear):
-                    trunc_normal_init(m.weight, std=.02)
-                    if m.bias is not None:
-                        constant_init(m.bias, 0)
+                    trunc_normal_init(m, std=.02, bias=0.)
                 elif isinstance(m, nn.LayerNorm):
                     constant_init(m.bias, 0)
```
Comment (Junjun2016) [inline mmdet/models/backbones/pvt.py:492]: `constant_init` is also used for `nn.Module`.

File: mmdet/models/backbones/pvt.py [hunk 5]
```diff
@@ -486,9 +487,7 @@ def init_weights(self):
                         f'training start from scratch')
             for m in self.modules():
                 if isinstance(m, nn.Linear):
-                    trunc_normal_init(m.weight, std=.02)
-                    if m.bias is not None:
-                        constant_init(m.bias, 0)
+                    trunc_normal_init(m, std=.02, bias=0.)
```
Comment (Junjun2016) [inline mmdet/models/backbones/pvt.py:490]: `normal_init` is also used for `nn.Module`.

**Pull request merged**